### PR TITLE
[FLINK-4108] [scala] Consider ResultTypeQueryable for input formats

### PR DESCRIPTION
**Pull request by twalthr**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

This PR fixes issues with input formats (i.e. the JDBC input format) in the Scala API. Now the API also considers explicit information defines with ResultTypeQueryable.

### CODE REVIEW

**Comment by albertoRamon** [comment]
Works OK  ¡¡
[Capture Output](http://apache-flink-user-mailing-list-archive.2336050.n4.nabble.com/jdbc-JDBCInputFormat-tp9393p9489.html)

Thanks, Alb

File: flink-scala/src/main/scala/org/apache/flink/api/scala/ExecutionEnvironment.scala [hunk 1]
```diff
@@ -110,7 +110,7 @@ class ExecutionEnvironment(javaEnv: JavaEnv) {
     */
   @PublicEvolving
   def getRestartStrategy: RestartStrategyConfiguration = {
-    javaEnv.getRestartStrategy()
+    javaEnv.getRestartStrategy
```
Comment (fhueske) [inline flink-scala/src/main/scala/org/apache/flink/api/scala/ExecutionEnvironment.scala]: We call methods on Java objects with parentheses.

**Comment by fhueske** [comment]
Reverting the change and merging

File: flink-scala/src/main/scala/org/apache/flink/api/scala/ExecutionEnvironment.scala (hunk 1 above)
Comment (fhueske) [inline flink-scala/src/main/scala/org/apache/flink/api/scala/ExecutionEnvironment.scala]: Apparently this is not the case in this class. Similar method are also called without parentheses. So lets keep this change to have consistent code in this class.

**Pull request closed without merging**